Lib/mailcap.py

```diff
@@ -19,6 +20,11 @@ def lineno_sort_key(entry):
     else:
         return 1, 0
 
+_find_unsafe = re.compile(r'[^\xa1-\U0010FFFF\w@%+=:,./-]').search
```

Comment:
is `%` safe? DOS derived shells use that for variable expansion IIRC.

Reply:
Mailcap implements the “Implementation Details for UNIX” appendix of RFC-1524 (which is what specifies using `%` for the placeholders), and that says:

    (Because of differences in shells and the implementation and behavior
    of the same shell from one system to another, it is specified that
    the command line be intended as input to the Bourne shell, i.e., that
    it is implicitly preceded by "/bin/sh -c " on the command line.)

So I'd say it's safe for Python to allow `%`. But it also won't hurt much to disallow it, so I'll do that.